### PR TITLE
Board files for LOLIN S3 MINI

### PR DESCRIPTION
**Pull request by tkroo**
My first time creating a new board definition, hopefully everything is ok.

### CODE REVIEW

File: ports/espressif/boards/lolin_s3_mini/board.c [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "supervisor/board.h"
+#include "mpconfigboard.h"
+#include "shared-bindings/microcontroller/Pin.h"
```
Comment (bill88t) [inline ports/espressif/boards/lolin_s3_mini/board.c:29]: You may want to copy over the debug no_resets from any other esp32s3 board.
They should be the same if the pins are exposed.

File: ports/espressif/boards/lolin_s3_mini/mpconfigboard.mk [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+USB_PRODUCT = "LOLIN S3 MINI 4MB Flash 2MB PSRAM"
+USB_MANUFACTURER = "WEMOS"
+
+IDF_TARGET = esp32s3
+
+CIRCUITPY_ESP_FLASH_MODE = qio
+CIRCUITPY_ESP_FLASH_FREQ = 80m
+CIRCUITPY_ESP_FLASH_SIZE = 4MB
+
+OPTIMIZATION_FLAGS = -Os
```
Comment (bill88t) [inline ports/espressif/boards/lolin_s3_mini/mpconfigboard.mk:12]: Why does this override optimisation?

File: ports/espressif/boards/lolin_s3_mini/sdkconfig [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+#
+# ESP32S3-Specific
+#
+CONFIG_ESP32S3_SPIRAM_SUPPORT=y
+#
+# SPI RAM config
+#
+# CONFIG_SPIRAM_MODE_QUAD=y
+# CONFIG_SPIRAM_MODE_OCT is not set
+CONFIG_SPIRAM_TYPE_AUTO=y
```
Comment (bill88t) [inline ports/espressif/boards/lolin_s3_mini/sdkconfig:13]: If you are sure of it being `QUAD`, you should try booting it without this.

**Comment by dhalbert** [comment]
@tkroo Does the latest batch of changes boot? If so I can approve. @bill88t How is it for you?

File: ports/espressif/boards/lolin_s3_mini/sdkconfig [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+# Component config
+#
+#
+# ESP32S3-Specific
+#
+CONFIG_ESP32S3_SPIRAM_SUPPORT=y
+#
+# SPI RAM config
+#
+# CONFIG_SPIRAM_MODE_QUAD=y
```
Comment (bill88t) [inline ports/espressif/boards/lolin_s3_mini/sdkconfig:11]: I should have been more specific, sorry.
It's `CONFIG_SPIRAM_TYPE_AUTO` that is unwanted.
You want `CONFIG_SPIRAM_MODE_QUAD` or `CONFIG_SPIRAM_MODE_OCT` enabled.

You can just try out if oct works if you are too lazy to check the schematic.
If oct doesn't boot, quad.

**Review by bill88t — COMMENTED** [review]
@dhalbert I do not have the board.
I just try to review stuff I am comfortable with.

The optimisation flag is what concerns me most with this definition.

If you want me to stop reviewing stuff, just tell me.

**Comment by dhalbert** [comment]
@bill88t No problem, keep reviewing. THe optimization change may be needed to make it fit. `-O2` is the default for the xtensa boards; `-Os` is needed for the RISC-V boards. (See the Makefile). You probably know this already. I don't see any other boards setting optimization per board in `ports/espressif`, though.

**Comment by bill88t** [comment]
Isn't it the policy to just disable unused modules in favor of keeping optimisation as is?

It's a 4mb S3 board.
Looking at the `adafruit_feather_esp32s3_tft` definition, instead of disabling optimisation, `CIRCUITPY_ESPCAMERA = 0` is set instead.

**Comment by dhalbert** [comment]
Normally we would turn ESPCAMERA off. We don't necessarily preserve optimization in favor of turning of general-purposes functionality. O2 is a lot bigger than Os. That is certainly true on various SAMD boards, for instance. We use Os to have all the useful modules.

**Comment by tkroo** [comment]
> @tkroo Does the latest batch of changes boot? If so I can approve. @bill88t How is it for you?

Yes, my board boots with latest changes. I also just did a build without the OPTIMIZATION_FLAG = -0s and it also boots. Should I remove that flag? I don't know what is preferred.

**Comment by bill88t** [comment]
If `OPTIMIZATION_FLAG` isn't needed, it should absolutely be removed.

**Review by bill88t — APPROVED** [review]

**Comment by tkroo** [comment]
> If `OPTIMIZATION_FLAG` isn't needed, it should absolutely be removed.

Thanks! Removed. I had thought the optimization flag was brought over from the existing Lolin S3 board files, but it was me who accidentally pasted it from somewhere else. Sorry about the confusion.

**Comment by skieast** [comment]
Although may not be relevant I would remove all the extraneous pin definitions for those not broken out on the board. I would include a definition for the onboard WS2812 which I believe is on IO47.  I can't be more specific on my phone lol 
Thanks

**Comment by dhalbert** [comment]
I don't understand why there are now merge conflicts noted above, since you are just adding files.

@tkroo, do you see @skieat's suggestion of removing pins that are not accessible?

**Comment by tkroo** [comment]
> @ktroo I don't understand why there are now merge conflicts noted, since you are just adding files.
> 
> Do you see @skieast's suggestion of removing pins that are not accessible?

I don't understand the merge conflicts either. Strange. But I am new to this whole process, so maybe I did something accidentally.
I did see @skieast's suggestions. On board neopixel is already defined:
https://github.com/adafruit/circuitpython/blob/5c4997850c84c8add0b2967acd99c0643d390521/ports/espressif/boards/lolin_s3_mini/pins.c#L101-L103
https://github.com/adafruit/circuitpython/blob/5c4997850c84c8add0b2967acd99c0643d390521/ports/espressif/boards/lolin_s3_mini/mpconfigboard.h#L32

re: extraneous pins. I checked pins.c and don't see any pins defined that are not also exposed on the board.

**Comment by dhalbert** [comment]
> I don't understand the merge conflicts either. Strange. But I am new to this whole process, so maybe I did something accidentally.

I think the easiest way to fix this would be to make a new branch, copy the new files into it, and make a new PR. We can close this one assuming the new one works.

**Comment by skieast** [comment]
> > @ktroo I don't understand why there are now merge conflicts noted, since you are just adding files.
> > Do you see @skieast's suggestion of removing pins that are not accessible?
> 
> I don't understand the merge conflicts either. Strange. But I am new to this whole process, so maybe I did something accidentally. I did see @skieast's suggestions. On board neopixel is already defined:
> 
> https://github.com/adafruit/circuitpython/blob/5c4997850c84c8add0b2967acd99c0643d390521/ports/espressif/boards/lolin_s3_mini/pins.c#L101-L103
> 
> 
> https://github.com/adafruit/circuitpython/blob/5c4997850c84c8add0b2967acd99c0643d390521/ports/espressif/boards/lolin_s3_mini/mpconfigboard.h#L32
> 
> re: extraneous pins. I checked pins.c and don't see any pins defined that are not also exposed on the board.
Quite true. I was looking at it on my phone. I can blame my old eyes, better than blaming myself.  Thanks again.

**Comment by tkroo** [comment]
> > I don't understand the merge conflicts either. Strange. But I am new to this whole process, so maybe I did something accidentally.
> 
> I think the easiest way to fix this would be to make a new branch, copy the new files into it, and make a new PR. We can close this one assuming the new one works.

Ok. I will create a new branch and PR. Thanks for all your help!

**Comment by tkroo** [comment]
> > re: extraneous pins. I checked pins.c and don't see any pins defined that are not also exposed on the board.
> 
> Quite true. I was looking at it on my phone. I can blame my old eyes, better than blaming myself. Thanks again.
No worries! I am new to this and appreciate as many eyes on this as possible. Thank you!

**Comment by dhalbert** [comment]
@tkroo you don't need to delete your fork. Just deleting the branch would be ifne.

**Comment by tkroo** [comment]
> @tkroo you don't need to delete your fork. Just deleting the branch would be ifne.

Oh well, I've already deleted it. I think what caused the merge conflict was that I had merged my branch into my fork's main. Anyway, I'll create a new fork, branch, and PR. 
Sorry about the chaos

**Comment by dhalbert** [comment]
> had merged my branch into my fork's main.

Yes, try to keep your `main` the same as upstream's `main`. Basically, never work in `main`. Always work in a branch.

**Comment by tkroo** [comment]
new PR: https://github.com/adafruit/circuitpython/pull/8023

**Pull request closed without merging**